readme.md
[![DOI](https://zenodo.org/badge/121652355.svg)](https://zenodo.org/badge/latestdoi/121652355)

This repository contains the complete scripts and data files used to construct the Computational History of Philosophy of Science (Comp HOPOS) dataset.  This readme file contains information on reproducing the dataset (and modifying the scripts for other purposes); a license statement; a brief overview of the method by which the dataset is constructed; and an overview of the files included in the repository.  

The dataset can be downloaded at <https://doi.org/10.5281/zenodo.1400633>.  The downloads include a data dictionary.  A paper describing the motivations and construction method in more detail is available at <link>.  

These scripts were developed by Daniel J. Hicks, Rick Morris, and Evelyn Brister.  The repository is maintained by Daniel J. Hicks.  To report errors or other issues, please use the issue tracker for this repository (preferred) or email <hicks.daniel.j@gmail.com>.  


# Reproducibility #

In principle, the dataset can be reproduced by running the scripts (the `R` files) in numerical order.  There are, of course, a number of further details and complications.  

- With the exception of the gender attribution websites accessed in `06_gender.R`, all of the software tools and data are free (or free for non-commercial use).  
- The `R` files are run using the [R programming language](https://cran.r-project.org/).  
- `04_dedupe_names.R` is essentially a wrapper around a separate (included) Python script.  This script requires [Python 3](https://www.python.org/downloads/).  The R script needs to be configured to call the correct Python installation.  
- The following input files are required: 
    1. `00_Boston.csv` and `00_Western_Ontario.csv`, which record volume-level information for the Boston and Western Ontario book series, respectively.  These files can be generated by conducting [a search on SpringerLink](https://link.springer.com/search?facet-series=%225710%22&facet-content-type=%22Book%22&showAll=true) and then clicking on the downward-arrow button in the upper-right.  
    2. `00_Minnesota.xlsx`, which records chapter-level information for the Minnesota book series.  This file was created by Evelyn Brister with assistance from Aaron Crespo.  
    3. `01_canonical_titles.csv`, which includes canonical names for journals.  This file can also be constructed manually from the output file `01_journal_data.csv`. 

- Every script has a number of dependencies, and the dependencies vary from script to script.  In particular, note the following: 
    - Unless otherwise noted, all dependencies should be available using standard package managers (e.g., R's built-in package manager, `pip` or Anaconda for Python).  
    - The Python dependencies used in `04_dupe_remove.py` are unique to that script.  
    - `06_gender.R` accesses files [in another repository](https://github.com/cblevins/Gender-ID-By-Time). 
    - `06_gender.R` also accesses APIs for two subscription services, [NamSor](http://www.namsor.com/) and [genderize.io](https://genderize.io/).  After subscribing to those services, create a file `api_keys.R` in the working directory, and add API keys with the following format:  
    ```
    namsor_key = 'stringofcharacters'
    namsor_user = 'namsor.com/e@mail.com/12345'
    genderize.io_key = 'otherstringofcharacters'
    ```

- The two scripts that retrieve larges amounts of data, `01_journals.R` and `02_springer_books.R`, have relatively long runtimes, approximately 20 and 45 minutes respectively.  

- Because of these two last points, it is recommended that each script be run individually.  While the entire process could be automated once all of the dependencies are satisfied, this will not be necessary in many cases (because the dataset has already been constructed).  

- Note on the CrossRef API from the RCRossRef documentation:  

> Be nice and share your email with Crossref
> 
> The Crossref team encourage requests with appropriate contact information and will forward you to a dedicated API cluster for improved performance when you share your email address with them. <https://github.com/CrossRef/rest-api-doc#good-manners--more-reliable-service>
> 
> To pass your email address to Crossref via this client, simply store it as environment variable in
.Renviron like this:
> 
> 1. Open file: file.edit("~/.Renviron")
> 2. Add email address to be shared with Crossref crossref_email = name@example.com
> 3. Save the file and restart your R session
> 
> Don’t wanna share your email any longer? Simply delete it from ~/.Renviron


# Copyright and License #

Copyright (c) 2018 Daniel J. Hicks, Rick Morris, and Evelyn Brister

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.


# Dataset Construction #

The primary source of data for the Comp HOPOS dataset is [CrossRef](https://en.wikipedia.org/wiki/Crossref), which maintains registration records for [the (vast) majority of digital object identifiers (DOIs)](https://github.com/greenelab/crossref/issues/3).  Recently, many scholarly publishers have been "minting" DOIs for their archives; combined with [an elegant R API](https://duckduckgo.com/?q=rcrossref&atb=v17&ia=web), this makes it possible to easily and rapidly retrieve a complete set of metadata records for many scholarly journals.  

Two other sources of data are incorporated into the Comp HOPOS dataset.  Chapter-level DOIs for the Boston and Western Ontario book series are scraped from search results in Springer's public search engine.  CrossRef is then used to retrieve the metadata for a chapter in a standard format.  For the Minnesota book series, Evelyn Brister retrieved the data from the University of Minnesota website manually, with assistance from Aaron Crespo.  

After combining these data sources, author names are disambiguated.  Using canonical names, philosophers of science are identified and genders are attributed to philosophers of science based on author names.  

Currently, philosophers of science are identified as authors who (after name disambiguation) have 2 or more articles in an identified set of "primary philosophy of science" journals (including the three book series).  


# Files Overview #

- `00_Boston.csv`: Volume-level information for Boston book series
- `00_Minnesota.xlsx`: Chapter-level information for the Minnesota book series
- `00_Western_Ontario.csv`: Volume-level information for the Western Ontario book series
- `01_canonical_titles.csv`: Canonical titles for journals
- `01_journals.R`: Retrieve metadata for journal articles using CrossRef API
- `02_springer_books.R`: Retrieve metadata for Springer-published book chapters using Springer website + CrossRef API
- `03_combine.R`: Combine data retrieved in previous two scripts
- `04_dedupe_names.R`: R wrapper around `04_dupe_remove.py`
- `04_dupe_remove.py`: Primary Python script used for author name disambiguation
- `04_qc_file_diffs.py`: Auxiliary Python script used for author name disambiguation
- `04_readme.txt`: Readme for Python scripts
- `05_verify_samples.R`: Generates subsets of author disambiguation output files for manual review
- `06_gender.R`:  Identifies philosophers of science in the dataset and attributes gender based on author name
- `07_dataset.R	`:  Combines metadataset with gender attribution into dataset suitable for public distribution
- `readme.md`:  This readme



NEWS
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).


## [Unreleased]

### Fixed
- The first run of Erkenntnis (prior to 1941) was treated as a primary philosophy of science venue when identifying philosophers of science (script 06), but not when assembling the dataset for release (script 07).  Erkenntnis is now handled correctly in both places.  This change moved 181 papers from "analytic" to "primary."  
- Missing values for `publication_series` for the book series.  [#14](https://github.com/dhicks/comp-HOPOS/issues/14)

### Added
- A Makefile in the `dataset_construction` folder.  In principle, once dependencies are satisfied and the necessary API keys are defined in `api_key.R`, the dataset can be reproduced from scratch by running `make` in this folder.  Due to the time involved to reproduce the dataset from scratch, this has not been fully tested.  
- `06_gender.R` now loads an old version of its output file `06_gender.Rds`, if present, and uses it as a cache for the gender attribution services.  This is mostly useful when upstream changes add, say, a hundred authors to the list of philosophers of science.  Set `force_query = TRUE` to force the script to re-query all author names. 

### Note
- There may be some kind of bug with UNF values in R 4.0.0.  <https://github.com/leeper/UNF/issues/19#issuecomment-650820112>
- Since the last update, NamSor has released version 2 of their API.  `06_gender.R` now calls version 2.  Combined with the caching of previous gender attribution, this may result in some inconsistencies in representation or value.  


## [2.0] - 2019-11-11

### Added
- This NEWS file
- Universal Numeric Fingerprints (UNF) are now used to support dataset validation.  The file `unf.csv` gives UNF hash strings for each dataset format, size, and file format.  By comparing these hash strings to working datasets, users can confirm which version of the dataset they are using.  
	- UNF are implemented using the `UNF` package in R.  <https://cran.r-project.org/web/packages/UNF/index.html>
	- For a brief introduction to UNF, see <https://cran.r-project.org/web/packages/UNF/vignettes/citation.html>
	- The following block briefly illustrates the use of UNF in practice:  
	
```{r}
library(UNF)

## UNF value for publications-philosophy of science-Rds v2.0
unf_value = 'nJaKSRjMpMV1zYGoOPFRlQ=='

pub_level = readRDS('publications_philsci.Rds')
pub_level_unf = unf(pub_level, version = 6, digits = 3, timezone = 'UTC')

identical(pub_level_unf$unf, unf_value)
```

### Removed
- Several redundant or (almost entirely) empty/NA columns were removed.  
	- Redundant `URL` column; cf <https://github.com/dhicks/comp-HOPOS/issues/11>
	- `member`, `prefix`, `score`, `source`, `subject`, `archive`, `authenticated.orcid`, `affiliation1.name`, `affiliation2.name`, `affiliation3.name`, `affiliation4.name`, `name`, `funder`, `assertion`
- Evelyn Brister manually identified and removed numerous non-article documents, such as tables of contents and book reviews. 
- Evelyn Brister manually identified authors who qualified as philosophers of science using the threshold criterion (i.e., 2 or more papers in a primary venue) but who primarily worked in other areas of philosophy.  These authors are:  
	- E. J. Lowe (metaphysics, phil mind, and phil lang.)
	- H B Acton (political philosophy)
	- Alasdair MacIntyre (ethics)
	- V. J. McGill
	- Jan Narveson (political theory)
	- Patrick Nowell-Smith (moral theory)
	- Daniel J O’Connor (philosophy of education)

## Fixed
- Evelyn Brister manually reviewed names and gender attribution, fixing issues related to initialization, misspellings, and incorrect or missing gender attribution (based on presentation on faculty websites, etc.).  
	- cf <https://github.com/dhicks/comp-HOPOS/issues/12>

## Changed
- The "philosophy of science" dataset size is now filtered by year, and includes only documents published between 1930 and 2017.  The first primary philosophy of science venue (the first version of *Erkenntnis*) began publication in 1930, so our approach identifies very few "philosophers of science" prior to this year.  



## [1.1] - 2018-08-26
### Fixed
This release fixes a substantial error that appeared when combing the gender attributions with the article metadata.

In v1.0, problems with the join logic when combining the results of the gender attribution algorithms (in script 06) meant that ~150 rows in the gender attribution dataframe had NA for both given and family names. All ~150 then matched to NA/NA author names in the article dataframe. The result was a massive inflation in the size of the dataset, and a mean of 26 authors per paper. Anyone familiar with philosophy should recognize this is incorrect.

Fixing the join logic in 06 appears to have solved the problem. Author inflation has disappeared. (In script 07, authors_unfltd has the same number of rows as authors_full.) In the full dataset, about 78% of papers have just 1 author; this is about 92% in the philosophy of science dataset.



scripts/06_gender.R
## Filter down to philosophers of science and attribute gender
##
## This script accesses files in the repository available at <https://github.com/cblevins/Gender-ID-By-Time>.  To use the script as configured, clone this script into the working directory.  
## This script also accesses APIs for two subscription services, [NamSor](http://www.namsor.com/) and [genderize.io](https://genderize.io/).  
## After subscribing to those services, create a file `api_keys.R` in the working directory, and add API keys with the following format:  
## ```
## namsor_key = 'stringofcharacters'
## namsor_user = 'namsor.com/e@mail.com/12345'
## genderize.io_key = 'otherstringofcharacters'
## ```

## Setup ----
library(tidyverse)

source('api_keys.R')

authors_df_unfltd = read_rds('../data/03_authors.rds') %>%
    filter(!is.na(family))
names_df = read_csv('../data/04_names_verif.csv', na = 'Ignored') %>%
    filter(!duplicated(.)) %>%
    mutate(`Canonical Family` = ifelse(is.na(`Canonical Family`), 
                                       `Orig Family`, 
                                       `Canonical Family`), 
           `Canonical Given` = ifelse(is.na(`Canonical Given`), 
                                      `Orig Given`, 
                                      `Canonical Given`))

## Combine author-level metadata and canonical names
authors_df = authors_df_unfltd %>%
    left_join(names_df, 
              by = c('family' = 'Orig Family', 
                     'given' = 'Orig Given')) %>%
    rename(family_orig = family, 
           given_orig = given) %>%
    ## Springer had some encoding errors that caused problems w/ deduping
    mutate(family = ifelse(!is.na(`Canonical Family`), 
                           `Canonical Family`, 
                           family_orig), 
           given = ifelse(!is.na(`Canonical Given`), 
                          `Canonical Given`, 
                          given_orig)) %>%
    ## Erkenntnis is primary prior to 1941
    mutate(publication_group = case_when(
        publication_series == 'Erkenntnis' & pub_year < 1941 ~ 'primary', 
        TRUE ~ publication_group))

## Filter down to "philosophers of science" 
phil_sci = authors_df %>%
    count(given, family, publication_group) %>%
    filter(publication_group == 'primary', n >= 2) %>%
    select(given, family) %>%
    inner_join(authors_df)

## Clean "given," removing initials and selecting what appears to be the first full name
phil_sci = phil_sci %>%
    select(given) %>%
    filter(!duplicated(.)) %>%
    mutate(split = str_split(given, ' ')) %>%
    unnest() %>%
    filter(!duplicated(.)) %>%
    mutate(len = str_length(split)) %>%
    filter(len > 1) %>%
    group_by(given) %>%
    filter(split == first(split)) %>%
    ungroup() %>%
    select(-len) %>%
    rename(for_gender_attr = split) %>%
    right_join(phil_sci)

phil_sci %>%
    select(given, family) %>%
    filter(!duplicated(.)) %>%
    write_rds('../data/06_phil_sci.Rds')

## Num. philosophy of science articles published in analytic journals
# phil_sci %>%
#     select(doi, container.title:pub_year,
#            publication_group, publication_series) %>%
#     filter(publication_group == 'analytic') %>%
#     filter(!duplicated(.)) %>%
#     ggplot(aes(pub_year)) +
#     geom_point(stat = 'count') +
#     geom_line(stat = 'count')

## Num. phil sci articles published across pub groups
# phil_sci %>%
#     select(doi, container.title:pub_year, 
#            publication_group, publication_series) %>%
#     filter(!duplicated(.)) %>%
#     count(pub_year, publication_group) %>%
#     ggplot(aes(pub_year, n, color = publication_group)) +
#     geom_line()

## Frac. phil sci articles across pub groups
## This is a nice one for showing that phil sci is gradually separating from analytic phil
# phil_sci %>%
#     select(doi, container.title:pub_year, 
#            publication_group, publication_series) %>%
#     filter(!duplicated(.)) %>%
#     count(pub_year, publication_group) %>%
#     group_by(pub_year) %>%
#     mutate(frac = n / sum(n)) %>%
#     ungroup() %>%
#     ggplot(aes(pub_year, frac, color = publication_group)) +
#     geom_line()


## Cameron Blevins: Gender ID by time ----
## <https://github.com/cblevins/Gender-ID-By-Time>

## Estimated yob is 30±5 years prior to first paper
author_first_pub = phil_sci %>%
    group_by(for_gender_attr, given, family) %>%
    summarize(first_pub = min(pub_year, na.rm = TRUE)) %>%
    ungroup() %>%
    mutate(yob_low = first_pub - 30 - 5, 
           yob_high = first_pub - 30 + 5)

## Load SSA YOB files
yob_folder = '../Gender-ID-By-Time/ReferenceData'
yob_files = list.files(yob_folder)
names(yob_files) = str_extract(yob_files, '[0-9]+')

## Takes a second
yob_df = yob_files %>%
    str_c(yob_folder, '/', .) %>% 
    `names<-`(names(yob_files)) %>%
    # head() %>%
    map(read_csv, col_names = c('given', 'gender', 'count')) %>%
    bind_rows(.id = 'yob') %>%
    rename(for_gender_attr = given)

## Takes a couple seconds
gender_blevins = author_first_pub %>%
    left_join(yob_df) %>%
    filter((yob_low <= yob & yob <= yob_high) | is.na(yob)) %>%
    group_by(for_gender_attr, given, family, gender) %>%
    summarize(count = sum(count)) %>%
    spread(gender, count, fill = 0) %>%
    select(-`<NA>`) %>%
    mutate(n = F+M, 
           prob_f_blevins = ifelse(n > 0, F / n, NA),
           gender_blevins = case_when(prob_f_blevins < .25 ~ 'm', 
                                      prob_f_blevins > .75 ~ 'f', 
                                      TRUE ~ 'indeterminate')) %>%
    ungroup()

# ggplot(gender_blevins, aes(prob_f)) + stat_ecdf() + geom_rug()
# 
# gender_blevins %>%
#     filter(prob_f > .25, prob_f < .75) %>% View
# 
# phil_sci %>%
#     left_join(gender_blevins) %>%
#     ggplot(aes(pub_year, fill = gender_blevins)) +
#     geom_bar(width = 1)
# 



## NamSor and nomine ----
## <https://github.com/cdcrabtree/nomine>
# library(nomine)
## Currently nomine raises an exception when the name contains spaces
## <https://github.com/cdcrabtree/nomine/issues/1>
# phil_sci %>%
#     count(given, family) %>%
#     sample_n(10) %>%
#     {with(., get_gender(given, family, 
#                         secret = namsor_key, user = namsor_user))}

## One-name-at-a-time version
namsor = function(given, family, namsor_key = NULL, namsor_user = NULL) {
    query_url = str_c('https://api.namsor.com/onomastics/api/json/gender/',
                      RCurl::curlEscape(given), '/', RCurl::curlEscape(family))
    full_url = str_c(query_url, '?key1=', namsor_key,
                     '&key2=', namsor_user)
    print(query_url)
    response = RCurl::getURL(full_url)
    json = jsonlite::fromJSON(response)
    json = jsonlite:::null_to_na(json)
    return(json)
}
# 
# phil_sci %>%
#     select(given, family) %>%
#     filter(!duplicated(.)) %>%
#     .[100:110,] %>%
#     transpose() %>%
#     map(~ namsor(.$given, .$family)) %>%
#     bind_rows()

## List/batched version
## NB API will raise an error w/ > 1k names
## ************ NB This can get expensive quickly! **************
## Make sure the namsor account is upgraded BEFORE running on a large list! 
namsor_list = function(names_df) {
    query_url = str_c('https://api.namsor.com/onomastics/api/json/genderList')
    header = c('Accept' = 'application/json', 
               'X-Channel-Secret' = namsor_key, 
               'X-Channel-User' = namsor_user)
    names_json = names_df %>%
        select(firstName = given, lastName = family) %>%
        filter(!duplicated(.)) %>%
        mutate(id = row_number()) %>%
        list('names' = .) %>%
        jsonlite::toJSON()
    
    response = RCurl::basicTextGatherer()
    result = RCurl::curlPerform(url = query_url, httpheader = header, postfields = names_json, writefunction = response$update)
    
    response_df = response$value() %>%
        jsonlite::fromJSON() %>%
        .$names %>%
        rename(given = firstName, family = lastName)
    return(response_df)
}

namsor_file = '../data/06_namsor.Rds'
if (!file.exists(namsor_file)) {
    chunks = phil_sci %>%
        count(given, family) %>%
        filter(complete.cases(.)) %>%
        mutate(row_num = row_number(), 
               chunk = as.integer(row_num %/% 1000)) %>%
        plyr::dlply('chunk', identity)
    
    gender_namsor = map_dfr(chunks, namsor_list)
    gender_namsor = gender_namsor %>%
        ## Rescale output variables
        mutate(gender = case_when(gender == 'male' ~ 'm', 
                                  gender == 'female' ~ 'f', 
                                  gender == 'unknown' ~ 'indeterminate'), 
               scale = (scale + 1)/2) %>%
        rename(gender_namsor = gender, 
               prob_f_namsor = scale)
    
    write_rds(gender_namsor, namsor_file)
} else {
    gender_namsor = read_rds(namsor_file)
}

ggplot(gender_namsor, aes(prob_f_namsor)) + 
    stat_ecdf() + geom_rug(aes(color = gender_namsor))



## genderize.io and genderizeR ----
# library(genderizeR)
## <https://github.com/kalimu/genderizeR>
## This is really designed for names embedded in blocks of text

# genderize = function (given, api_key = NULL) {
#     query_url = str_c('https://api.genderize.io/?name=', given)
#     if (!is.null(api_key)) {
#         query_url = str_c(query_url, '&apikey=', api_key)
#     }
#     response = RCurl::getURL(query_url)
#     json = jsonlite::fromJSON(response, simplifyDataFrame = TRUE)
#     json = jsonlite:::null_to_na(json)
#     return(json)
# }
# ## 8.793 sec / 25 names -> ~14 minutes for all 2384 names
# tictoc::tic()
# thing = phil_sci %>%
#     filter(!is.na(for_gender_attr)) %>%
#     pull(for_gender_attr) %>%
#     unique() %>%
#     .[1:25] %>%
#     map_dfr(genderize) %>%
#     rename(for_gender_attr = name)
# tictoc::toc()

## Batch/list version
genderize_list = function(given, api_key = NULL) {
    names = str_c('name[]=', given, collapse = '&')
    query_url = str_c('https://api.genderize.io/?', names)
    if (!is.null(api_key)) {
        query_url = str_c(query_url, '&apikey=', api_key)
    }
    response = RCurl::getURL(query_url)
    json = jsonlite::fromJSON(response, simplifyDataFrame = TRUE)
    json = jsonlite:::null_to_na(json)
    return(json)
}

chunks_genderize = phil_sci %>%
    filter(!is.na(for_gender_attr)) %>%
    count(for_gender_attr) %>%
    mutate(chunk = row_number() %/% 10) %>%
    plyr::dlply('chunk', identity) %>%
    map(pull, for_gender_attr)

## 1.042 sec / 3 chunks -> ~45 sec for all 128 chunks
# tictoc::tic()
# map(chunks_genderize[1:3], genderize_list)
# tictoc::toc()

genderize_file = '../data/06_genderize.Rds'
if (!file.exists(genderize_file)) {
    # tictoc::tic()
    gender_genderize = chunks_genderize %>%
        map_dfr(genderize_list, api_key = genderize.io_key)
    # tictoc::toc()
    
    gender_genderize = gender_genderize %>%
        ## Rescale output variables
        mutate(gender = case_when(gender == 'male' ~ 'm',
                                  gender == 'female' ~ 'f',
                                  is.na(gender) ~ 'indeterminate'),
               probability = ifelse(gender == 'm', 1-probability, probability)) %>%
        rename(prob_f_genderize = probability, 
               gender_genderize = gender,
               for_gender_attr = name)
    
    write_rds(gender_genderize, genderize_file)
} else {
    gender_genderize = read_rds(genderize_file)
}


## Combine ----

gender_combined = phil_sci %>%
    select(for_gender_attr, given, family) %>%
    filter(!duplicated(.)) %>%
    left_join(gender_blevins) %>% 
    select(-`F`, -M, -n) %>%
    left_join(gender_namsor) %>%
    select(-id) %>%
    left_join(gender_genderize) %>%
    select(-count) %>% 
    rowwise() %>%
    mutate(avg = mean(c(prob_f_blevins, 
                        prob_f_namsor,
                        prob_f_genderize), 
                      na.rm = TRUE)) %>%
    ungroup() %>%
    mutate(gender_attr = case_when(avg < .25 ~ 'm', 
                                   avg > .75 ~ 'f', 
                                   TRUE ~ 'indet'))

write_rds(gender_combined, '../data/06_gender.Rds')

# ggplot(gender_combined, aes(prob_f_blevins, prob_f_namsor)) + 
#     geom_point() +
#     ggrepel::geom_label_repel(aes(label = str_c(given, ' ', family)), 
#                               data = function (dataf) dataf[abs(dataf$prob_f_blevins - dataf$prob_f_namsor) > .5,]) +
#     theme_bw()

ggplot(gender_combined, aes(avg)) + stat_ecdf()
filter(gender_combined, gender_attr == 'indet') %>% 
    write_csv('../data/06_indeterminate_gender.csv')





